components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java

```diff
@@ -0,0 +1,851 @@
+import static org.wso2.carbon.identity.organization.management.service.util.Utils.getAuthenticatedUsername;
+import static org.wso2.carbon.identity.organization.management.service.util.Utils.getTenantDomain;
+import static org.wso2.carbon.identity.organization.management.service.util.Utils.getTenantId;
+import static org.wso2.carbon.identity.organization.management.service.util.Utils.getUserId;
+import static org.wso2.carbon.identity.organization.management.service.util.Utils.handleClientException;
+import static org.wso2.carbon.identity.organization.management.service.util.Utils.handleServerException;
+
+/**
+ * This class implements the {@link OrganizationManager} interface.
+ */
+public class OrganizationManagerImpl implements OrganizationManager {
+
+    private final OrganizationManagementDAO organizationManagementDAO = new OrganizationManagementDAOImpl();
+
+    @Override
+    public Organization addOrganization(Organization organization) throws OrganizationManagementException {
+
+        validateAddOrganizationRequest(organization);
+        setParentOrganization(organization);
+        setCreatedAndLastModifiedTime(organization);
+        organizationManagementDAO.addOrganization(organization);
+        String orgCreatorID = getUserId();
+        String orgCreatorName = getAuthenticatedUsername();
+        if (StringUtils.equals(TENANT.toString(), organization.getType())) {
+            createTenant(organization.getId(), orgCreatorID, orgCreatorName);
+        }
+        return organization;
+    }
+
+    @Override
+    public boolean isOrganizationExistByName(String organizationName) throws OrganizationManagementException {
+
+        return organizationManagementDAO.isOrganizationExistByName(organizationName);
+    }
+
+    @Override
+    public boolean isOrganizationExistById(String organizationId) throws OrganizationManagementException {
+
+        return organizationManagementDAO.isOrganizationExistById(organizationId);
+    }
+
+    @Override
+    public String getOrganizationIdByName(String organizationName) throws OrganizationManagementException {
+
+        return organizationManagementDAO.getOrganizationIdByName(organizationName);
+    }
+
+    @Override
+    public String getOrganizationNameById(String organizationId) throws OrganizationManagementException {
+
+        return organizationManagementDAO.getOrganizationNameById(organizationId).orElseThrow(
+                () -> handleClientException(ERROR_CODE_INVALID_ORGANIZATION_ID, organizationId));
+    }
+
+    @Override
+    public Organization getOrganization(String organizationId, boolean showChildren, boolean includePermissions) throws
+            OrganizationManagementException {
+
+        if (StringUtils.isBlank(organizationId)) {
+            throw handleClientException(ERROR_CODE_ORGANIZATION_ID_UNDEFINED);
+        }
+        Organization organization = organizationManagementDAO.getOrganization(organizationId.trim());
+
```

Comment:
We can improve the readability by moving the new line in 193 to 192

Reply:
resolved by 6534bc4a65f7a5c7396abda24169fb0e5690702a